### ISSUE TITLE
Hourglass crash on negative values

### ISSUE THREAD

**Issue by daonb** [tuzig/terminal7#344]
**Describe the bug**
It's a known issue with progress bar - it's hard to keep them positive.
There are always rear edge cases when they go off the charts.

I got an error: [Error] RangeError: String.prototype.repeat argument must be greater than or equal to 0 and not be Infinity 

in src/shell.ts on the line:

```
this.t.write(`\r\x1B[KTWR ${" ".repeat(i)}ᗧ${"·".repeat(len-i-1)}🍒\x1B[?25l`)
```

The debugger told me both `len` and `i` are 20.

**To Reproduce**
Not sure

**Screenshots**

**Additional context**
Simplest solution is to calculate it separatly and make sure it's not negative

**Comment by eyal-sasson** [comment]
I'll just implement a simple fix for this in #338 although it really shouldn't happen as the watchdog stops the hourglass after the given timeout.

**Comment by daonb** [comment]
It shouldn't happen, but that's software for you.
